subsys/dfu/boot/mcuboot.c

```diff
@@ -151,24 +150,23 @@ int mcuboot_swap_type_multi(int image_index)
 
 int mcuboot_swap_type(void)
 {
-#ifdef FLASH_AREA_IMAGE_SECONDARY
+#if !ZEPHYR_MCUBOOT_APP_IS_SINGLE_SLOT
```

Comment:
This is where the problem is going to arise, this should be controlled by Kconfig, not the automatic check in the header. Kconfig selection should be limited to single slot only if there is only one slot, but the user should be free to select single slot or dual slot modes depending upon their application needs

Reply:
Removed the ZEPHYR_MCUBOOT_APP_IS_SINGLE_SLOT